### ISSUE TITLE
auto allows functions to start after error is passed

### ISSUE THREAD

**Issue by e174596549** [suguru03/neo-async#57]
This issue [#988](https://github.com/caolan/async/issues/988) was fixed in async v1.5.1 but still exist in neo-async.

Throw an exception when async.auto calls back with an error `Error: async.auto task has cyclic dependencies`.

**Comment by suguru03** [comment]
Thanks for the issue!

I have the test case and can't reproduce it.
Which version are you using? Also, do you have any code to reproduce it?

https://github.com/suguru03/neo-async/blob/3cfae1dcbe20442c99d7d82115baa4877be6f75a/test/controlFlow/test.auto.js#L320-L334

**Comment by e174596549** [comment]
neo-async v2.5

```
const async = require('neo-async');

async.auto({
    task1: function(callback) {
        setTimeout(()=>{callback('error')}, 1000)
    },
    task2: function(callback) {
        setTimeout(()=>{callback(false)}, 2000)
    },
    task3: ['task1', 'task2', function(callback) {
        setTimeout(()=>{callback(false)}, 2000)
    }]
}, function(err) {
    if(err) {
        return console.log('err: ', err)
    }
    console.log('done')
});

/usr/local/bin/node /Users/neo-async-test.js
err:  error
/Users/node_modules/.2.5.0@neo-async/async.js:8167
          throw new Error('async.auto task has cyclic dependencies');
          ^

Error: async.auto task has cyclic dependencies
    at proceedQueue (/Users/node_modules/.2.5.0@neo-async/async.js:8167:17)
    at taskComplete (/Users/node_modules/.2.5.0@neo-async/async.js:8187:7)
    at done (/Users/node_modules/.2.5.0@neo-async/async.js:8153:9)
    at Timeout.setTimeout [as _onTimeout] (/Users/neo-async-test.js:8:25)
    at ontimeout (timers.js:475:11)
    at tryOnTimeout (timers.js:310:5)
    at Timer.listOnTimeout (timers.js:270:5)

```

async v2.1.4

```
const async = require('async');

async.auto({
    task1: function(callback) {
        setTimeout(()=>{callback('error')}, 1000)
    },
    task2: function(callback) {
        setTimeout(()=>{callback(false)}, 2000)
    },
    task3: ['task1', 'task2', function(callback) {
        setTimeout(()=>{callback(false)}, 2000)
    }]
}, function(err) {
    if(err) {
        return console.log('err: ', err)
    }
    console.log('done')
});


err:  error

Process finished with exit code 0
```

**Comment by suguru03** [comment]
@e174596549 
I can reproduce it!
I'll fix it soon, thanks! 😄

**Comment by suguru03** [comment]
I published it on `v2.5.1`, could you check it? Thanks! 😸
